### PR TITLE
cleanup: reuse code logic in NodeInfo.AddPod & NodeInfo.RemovePod

### PR DESCRIPTION
**Pull request by binacs**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Reuse part of the similar code.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/types.go [hunk 1]
```diff
@@ -708,6 +675,26 @@ func (n *NodeInfo) RemovePod(pod *v1.Pod) error {
...
+	if n.Requested.ScalarResources == nil && len(res.ScalarResources) > 0 {
+		n.Requested.ScalarResources = map[v1.ResourceName]int64{}
+	}
+	for rName, rQuant := range res.ScalarResources {
+		n.Requested.ScalarResources[rName] += sign * rQuant
+	}
+	n.NonZeroRequested.MilliCPU += sign * non0CPU
+	n.NonZeroRequested.Memory += sign * non0Mem
+
+	n.updateUsedPorts(pod, sign > 0)
```
Comment (kerthcet) [inline pkg/scheduler/framework/types.go:694]: Let's add back the comments?

Comment (kerthcet) [reply pkg/scheduler/framework/types.go:694]: Or rename the method name to `update()` and add `n.Generation = nextGeneration()` to this method too. At the least, let's add back the comments for updateUsedPorts. WDYT?

**Comment by kerthcet** [comment]
one nit
/hold

File: pkg/scheduler/framework/types.go (hunk 1 above)
Comment (BinacsLee) [reply pkg/scheduler/framework/types.go:693]: Make sense, addressed comments. PATL

**Comment by BinacsLee** [comment]
kindly ping @alculquicondor for approval

File: pkg/scheduler/framework/types.go [hunk 2]
```diff
@@ -681,33 +664,38 @@ func (n *NodeInfo) RemovePod(pod *v1.Pod) error {
...
-			n.Generation = nextGeneration()
+			n.update(pod, -1)
 			n.resetSlicesIfEmpty()
 			return nil
 		}
 	}
 	return fmt.Errorf("no corresponding pod %s in pods of node %s", pod.Name, n.node.Name)
 }
 
+// Update node info based on the pod and sign.
```
Comment (alculquicondor) [inline pkg/scheduler/framework/types.go:676]:
```suggestion
// update node info based on the pod and sign.
```

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by alculquicondor** [comment]
/hold
for nit

File: pkg/scheduler/framework/types.go (hunk 2 above)
Comment (BinacsLee) [reply pkg/scheduler/framework/types.go]: @alculquicondor fixed, PTAL

**Pull request merged**